benchmark/single-source/StringReplaceSubrange.swift

```diff
@@ -0,0 +1,116 @@
+//===--- StringReplaceSubrange.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+let tags: [BenchmarkCategory] = [.validation, .api, .String]
+
+public let StringReplaceSubrange = [
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.String",
+    runFunction: { replaceSubrange($0, "coffee", with: "t") },
+    tags: tags
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.LargeManaged.String",
+    runFunction: { replaceSubrange($0, largeManagedString, with: "t") },
+    tags: tags,
+    setUpFunction: setupLargeManagedString
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.Substr",
+    runFunction: { replaceSubrange($0, "coffee", with: getSubstring("t")) },
```

Comment:
We don't need to put an optimization barrier (by calling the `getSubstring` from `TestUtils`) here.
Shorter way to get a `Substring` is to get a full subrange like this: `"t"[...]`.

For an implementation symmetry, I'd also extract the "coffee" into `smallString` constant, so that our benchmark definitions would vary only like this:
* `runFunction: { replaceSubrange($0, largeString, with: "t") }`
* `runFunction: { replaceSubrange($0, smallString, with: "t") }`
* `runFunction: { replaceSubrange($0, largeString, with: "t"[...]) }`
* `runFunction: { replaceSubrange($0, smallString, with: "t"[...]) }`